src/base_home.cpp

```diff
@@ -0,0 +1,94 @@
+#include "base_home.h"
+#include "submap.h"
+#include <unordered_map>
+#include "mapdata.h"
+#include "map.h"
+
+base_home::base(submap &base_map,const tripoint &coreloc){
+    base::define_base_area(base_map, coreloc);
+    base_map.spawns.clear();//remove spawns so nothing apears in base just outside it.
+    base_level = 1;
+    food_ration_lv = ammo_ration_lv = med_ration_lv = 0; //no rationing to start
+}
+
+/**
+ * Check if location has a base specific flag.
+ */
+bool base_home::has_base_flag(const tripoint &p, base_area_flag flag)
+{
+    return(baflag[p.x][p.y])
```

Comment:
The function ignores the `flag` parameter. Why?

Reply:
I wasn't thinking straight?